src/world/Objects/Units/Creatures/Creature.cpp

```diff
@@ -884,15 +884,9 @@ void Creature::SaveToDB()
         m_spawn->channel_spell = 0;
         m_spawn->MountedDisplayID = getMountDisplayId();
 
-#if VERSION_STRING < WotLK
-        m_spawn->Item1SlotEntry = getVirtualItemEntry(MELEE);
-        m_spawn->Item2SlotEntry = getVirtualItemEntry(OFFHAND);
-        m_spawn->Item3SlotEntry = getVirtualItemEntry(RANGED);
-#else
-        m_spawn->Item1SlotEntry = getVirtualItemSlotId(MELEE);
-        m_spawn->Item2SlotEntry = getVirtualItemSlotId(OFFHAND);
-        m_spawn->Item3SlotEntry = getVirtualItemSlotId(RANGED);
-#endif
+        m_spawn->itemEquipSlots[MELEE] = getVirtualItemSlotId(MELEE);
```

Comment:
```c
#if VERSION_STRING >= WotLK
uint32_t Unit::getVirtualItemSlotId(uint8_t slot) const { return unitData()->virtual_item_slot_display[slot]; }
#else
uint32_t Unit::getVirtualItemDisplayId(uint8_t slot) const { return unitData()->virtual_item_slot_display[slot]; }
#endif```
there is nearly no diff in code algorithm

Reply:
Nope, but wowdata (= clientside data) value _virtual_item_slot_display_ stores item display id in Classic/TBC and item entry since Wotlk. What I meant here is to use getVirtualItemEntry in Classic/TBC and getVirtualItemSlotId in other versions like it used to be.